### PR TITLE
Makes the HomeActivity a lifecycle owner

### PR DESCRIPTION
**Pull request by florina-muntenescu**

### CODE REVIEW

**Comment by florina-muntenescu** [comment]
This PR contains the fixes from #622 as well.

**Comment by florina-muntenescu** [comment]
@ZacSweers any change you can take a look at this Dagger related PR? Thanks a lot in advance!

File: app/src/main/java/io/plaidapp/dagger/HomeModule.kt [hunk 1]
```diff
@@ -40,20 +43,30 @@ import io.plaidapp.core.dribbble.data.api.model.Shot
...
 
-    @Provides
-    fun activity(): Activity = activity
+    @Binds
+    abstract fun homeActivityAsActivity(activity: HomeActivity): Activity
 
-    @Provides
-    fun columns(): Int = activity.resources.getInteger(R.integer.num_columns)
+    @Binds
+    abstract fun context(activity: Activity): Context
```
Comment (keyboardsurfer) [inline app/src/main/java/io/plaidapp/dagger/HomeModule.kt]: Looks like we could remove this one. At least I don't see any references to raw `context`

Comment (florina-muntenescu) [reply app/src/main/java/io/plaidapp/dagger/HomeModule.kt]: It's needed for `SharedPreferences`.

**Pull request merged**